In [86]:
#we load the data 
import pandas as pd
import numpy as np
import math
from scipy.stats import ttest_ind
data=pd.read_stata("../RawData/ZA7745_MannheimCorona_GIP/ZA7745_v1-0-0.dta")
data

za_nr             version                 doi        id_g    hhid_g  \
32     7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  10020401.0  100204.0   
33     7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  10020401.0  100204.0   
34     7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  10020401.0  100204.0   
35     7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  10020401.0  100204.0   
36     7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  10020401.0  100204.0   
...       ...                 ...                 ...         ...       ...   
70043  7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  49934901.0  499349.0   
70044  7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  49934901.0  499349.0   
70045  7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  49934901.0  499349.0   
70046  7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  49934901.0  499349.0   
70047  7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  49934901.0  499349.0   

       pid      dDatum              exp_group  Woche Teilnahme  ...  \
32     1.0  23.03.2020  3. Gruppe 3 (Sonntag)    1.0     1. ja  ...   
33     1.0  29.03.2020  3. Gruppe 3 (Sonntag)    2.0     1. ja  ...   
34     1.0  05.04.2020  3. Gruppe 3 (Sonntag)    3.0     1. ja  ...   
35     1.0  12.04.2020  3. Gruppe 3 (Sonntag)    4.0     1. ja  ...   
36     1.0  19.04.2020  3. Gruppe 3 (Sonntag)    5.0     1. ja  ...   
...    ...         ...                    ...    ...       ...  ...   
70043  1.0  08.06.2020  3. Gruppe 3 (Sonntag)   12.0     1. ja  ...   
70044  1.0  14.06.2020  3. Gruppe 3 (Sonntag)   13.0     1. ja  ...   
70045  1.0  21.06.2020  3. Gruppe 3 (Sonntag)   14.0     1. ja  ...   
70046  1.0  28.06.2020  3. Gruppe 3 (Sonntag)   15.0     1. ja  ...   
70047  1.0  06.07.2020  3. Gruppe 3 (Sonntag)   16.0     1. ja  ...   

                                     SCPX032  \
32                                       NaN   
33                                       NaN   
34                                       NaN   
35                                       NaN   
36                                       NaN   
...                                      ...   
70043                                    NaN   
70044                                    NaN   
70045                                    NaN   
70046  10. voll und ganz verantwortlich sein   
70047                                    NaN   

                                     SCPX033  \
32                                       NaN   
33                                       NaN   
34                                       NaN   
35                                       NaN   
36                                       NaN   
...                                      ...   
70043                                    NaN   
70044                                    NaN   
70045                                    NaN   
70046  10. voll und ganz verantwortlich sein   
70047                                    NaN   

                                     SCPX034       SCPX035            SCPX036  \
32                                       NaN           NaN                NaN   
33                                       NaN           NaN                NaN   
34                                       NaN           NaN                NaN   
35                                       NaN           NaN                NaN   
36                                       NaN           NaN                NaN   
...                                      ...           ...                ...   
70043                                    NaN           NaN                NaN   
70044                                    NaN           NaN                NaN   
70045                                    NaN           NaN                NaN   
70046  10. voll und ganz verantwortlich sein  2. Stimme zu  3. Kommt drauf an   
70047                                    NaN           NaN                NaN   

                  SCPX037             SCPX038                       SCPX039  \


In [75]:
#check the number of individuals in the dataset
len(data["id_g"].unique())

4387

In [76]:
#and check which variable is at which position of the columns

a=0
for i in data.columns:
    print(a)
    print(i)
    a+=1


0
za_nr
1
version
2
doi
3
id_g
4
hhid_g
5
pid
6
dDatum
7
exp_group
8
Woche
9
Teilnahme
10
sample
11
educ_school_19
12
educ_job_19
13
marital_status_19
14
number_hh_members_19
15
german_citizenship_19
16
internet_usage_19
17
state_19
18
occupation_19
19
year_of_birth_cat_19
20
gender_19
21
state_residence
22
state_today
23
SCFX001
24
SCFX001_1
25
SCFX002_a
26
SCFX002_b
27
SCFX002_c
28
SCFX002_d
29
SCFX002_e
30
SCFX003_a
31
SCFX003_b
32
SCFX003_c
33
SCFX003_d
34
SCFX003_e
35
SCFX003_f
36
SCFX003_g
37
SCFX003_h
38
SCDX001
39
SCDX004
40
SCBX001
41
SCBX002
42
SCPX001
43
SCPX002
44
SCPX003
45
SCPX004
46
SCPX005
47
SCPX006_a
48
SCPX006_b
49
SCPX006_c
50
SCPX006_d
51
SCPX006_e
52
SCPX006_f
53
SCPX006_g
54
SCBX003
55
SCBX004_1
56
SCBX004_2
57
SCBX004_3
58
SCBX004_4
59
SCBX005
60
SCBX005_1
61
SCBX005_2
62
SCBX006
63
SCBX007
64
SCBX008
65
SCBX009
66
SCBX010
67
SCBX011
68
SCPX007
69
SCPX007_1
70
SCBX012
71
SCDX005
72
SCBX013_1
73
SCBX013_2
74
SCBX013_3
75
SCBX013_4
76
SCBX013_5
77
SCBX013_6
78
SCB

In [77]:
#so we save all data up to the 22nd column as identifiers 
extract=data.iloc[:,:22]

In [78]:
#we extract whether the individuals have a partner or not
extract["partner"]=np.where(data.SCFX004=="1. Ja","1","")
extract["partner"]=np.where(data.SCFX004=="2. Nein","0",extract["partner"])
extract

za_nr             version                 doi        id_g    hhid_g  \
0      7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  10005901.0  100059.0   
1      7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  10005901.0  100059.0   
2      7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  10005901.0  100059.0   
3      7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  10005901.0  100059.0   
4      7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  10005901.0  100059.0   
...       ...                 ...                 ...         ...       ...   
70187  7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  49988201.0  499882.0   
70188  7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  49988201.0  499882.0   
70189  7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  49988201.0  499882.0   
70190  7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  49988201.0  499882.0   
70191  7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  49988201.0  499882.0   

       pid      dDatum              exp_group  Woche Teilnahme  ...  \
0      1.0  21.03.2020  2. Gruppe 2 (Samstag)    1.0     1. ja  ...   
1      1.0  28.03.2020  2. Gruppe 2 (Samstag)    2.0     1. ja  ...   
2      1.0  04.04.2020  2. Gruppe 2 (Samstag)    3.0     1. ja  ...   
3      1.0  11.04.2020  2. Gruppe 2 (Samstag)    4.0     1. ja  ...   
4      1.0  18.04.2020  2. Gruppe 2 (Samstag)    5.0     1. ja  ...   
...    ...         ...                    ...    ...       ...  ...   
70187  1.0  08.06.2020   4. Gruppe 4 (Montag)   12.0     1. ja  ...   
70188  1.0  15.06.2020   4. Gruppe 4 (Montag)   13.0     1. ja  ...   
70189  1.0  22.06.2020   4. Gruppe 4 (Montag)   14.0     1. ja  ...   
70190  1.0  29.06.2020   4. Gruppe 4 (Montag)   15.0     1. ja  ...   
70191  1.0  06.07.2020   4. Gruppe 4 (Montag)   16.0     1. ja  ...   

                                       marital_status_19 number_hh_members_19  \
0                                               5. Ledig                 1. 1   
1                                               5. Ledig                 1. 1   
2                                               5. Ledig                 1. 1   
3                                               5. Ledig                 1. 1   
4                                               5. Ledig                 1. 1   
...                                                  ...                  ...   
70187  1. Verheiratet und leben mit Ihrem/Ihrer Ehepa...                 2. 2   
70188  1. Verheiratet und leben mit Ihrem/Ihrer Ehepa...                 2. 2   
70189  1. Verheiratet und leben mit Ihrem/Ihrer Ehepa...                 2. 2   
70190  1. Verheiratet und leben mit Ihrem/Ihrer Ehepa...                 2. 2   
70191  1. Verheiratet und leben mit Ihrem/Ihrer Ehepa...                 2. 2   

                         german_citizenship_19 internet_usage_19  \
0      1. Ja, nur die deutsche Staatsangehörig        8. Täglich   
1      1. Ja, nur die deutsche Staatsangehörig        8. Täglich   
2      1. Ja, nur die deutsche Staatsangehörig        8. Täglich   
3      1. Ja, nur die deutsche Staatsangehörig        8. Täglich   
4      1. Ja, nur die deutsche Staatsangehörig        8. Täglich   
...                                        ...               ...   
70187  1. Ja, nur die deutsche Staatsangehörig        8. Täglich   
70188  1. Ja, nur die deutsche Staatsangehörig        8. Täglich   
70189  1. Ja, nur die deutsche Staatsangehörig        8. Täglich   
70190  1. Ja, nur die deutsche Staatsangehörig        8. Täglich   
70191  1. Ja, nur die deutsche Staatsangehörig        8. Täglich   

                      state_19            occupation_19 year_of_birth_cat_19  \
0      4. Niedersachsen/Bremen  1. Vollzeiterwerbstätig         7. 1965-1969   
1      4. Niedersachsen/Bremen  1. Vollzeiterwerbstätig         7. 1965-1969   
2      4. Niedersachsen/Bremen  1. Vollzeiterwerbstätig         7. 1965-1969   
3      4. Niedersachsen/Bremen  1. Vollzeiterwerbstätig         7. 1965-1969   
4      4. Nieder

In [79]:
#how often did they meet others before and since the pandemic
extract["Treffen_vorher"]=data["SCBX001"]
extract["Treffen_nachher"]=data["SCBX002"]
extract

za_nr             version                 doi        id_g    hhid_g  \
0      7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  10005901.0  100059.0   
1      7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  10005901.0  100059.0   
2      7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  10005901.0  100059.0   
3      7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  10005901.0  100059.0   
4      7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  10005901.0  100059.0   
...       ...                 ...                 ...         ...       ...   
70187  7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  49988201.0  499882.0   
70188  7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  49988201.0  499882.0   
70189  7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  49988201.0  499882.0   
70190  7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  49988201.0  499882.0   
70191  7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  49988201.0  499882.0   

       pid      dDatum              exp_group  Woche Teilnahme  ...  \
0      1.0  21.03.2020  2. Gruppe 2 (Samstag)    1.0     1. ja  ...   
1      1.0  28.03.2020  2. Gruppe 2 (Samstag)    2.0     1. ja  ...   
2      1.0  04.04.2020  2. Gruppe 2 (Samstag)    3.0     1. ja  ...   
3      1.0  11.04.2020  2. Gruppe 2 (Samstag)    4.0     1. ja  ...   
4      1.0  18.04.2020  2. Gruppe 2 (Samstag)    5.0     1. ja  ...   
...    ...         ...                    ...    ...       ...  ...   
70187  1.0  08.06.2020   4. Gruppe 4 (Montag)   12.0     1. ja  ...   
70188  1.0  15.06.2020   4. Gruppe 4 (Montag)   13.0     1. ja  ...   
70189  1.0  22.06.2020   4. Gruppe 4 (Montag)   14.0     1. ja  ...   
70190  1.0  29.06.2020   4. Gruppe 4 (Montag)   15.0     1. ja  ...   
70191  1.0  06.07.2020   4. Gruppe 4 (Montag)   16.0     1. ja  ...   

                         german_citizenship_19 internet_usage_19  \
0      1. Ja, nur die deutsche Staatsangehörig        8. Täglich   
1      1. Ja, nur die deutsche Staatsangehörig        8. Täglich   
2      1. Ja, nur die deutsche Staatsangehörig        8. Täglich   
3      1. Ja, nur die deutsche Staatsangehörig        8. Täglich   
4      1. Ja, nur die deutsche Staatsangehörig        8. Täglich   
...                                        ...               ...   
70187  1. Ja, nur die deutsche Staatsangehörig        8. Täglich   
70188  1. Ja, nur die deutsche Staatsangehörig        8. Täglich   
70189  1. Ja, nur die deutsche Staatsangehörig        8. Täglich   
70190  1. Ja, nur die deutsche Staatsangehörig        8. Täglich   
70191  1. Ja, nur die deutsche Staatsangehörig        8. Täglich   

                      state_19            occupation_19 year_of_birth_cat_19  \
0      4. Niedersachsen/Bremen  1. Vollzeiterwerbstätig         7. 1965-1969   
1      4. Niedersachsen/Bremen  1. Vollzeiterwerbstätig         7. 1965-1969   
2      4. Niedersachsen/Bremen  1. Vollzeiterwerbstätig         7. 1965-1969   
3      4. Niedersachsen/Bremen  1. Vollzeiterwerbstätig         7. 1965-1969   
4      4. Niedersachsen/Bremen  1. Vollzeiterwerbstätig         7. 1965-1969   
...                        ...                      ...                  ...   
70187   5. Nordrhein-Westfalen  1. Vollzeiterwerbstätig        11. 1985-1989   
70188   5. Nordrhein-Westfalen  1. Vollzeiterwerbstätig        11. 1985-1989   
70189   5. Nordrhein-Westfalen  1. Vollzeiterwerbstätig        11. 1985-1989   
70190   5. Nordrhein-Westfalen  1. Vollzeiterwerbstätig        11. 1985-1989   
70191   5. Nordrhein-Westfalen  1. Vollzeiterwerbstätig        11. 1985-1989   

         gender_19          state_residence partner  \
0      2. weiblich  4. Niedersachsen/Bremen           
1      2. weiblich  4. Niedersachsen/Bremen           
2      2. weiblich  4. Niedersachsen/Bremen           
3      2. weiblich  4. Niedersachsen/Bremen           
4      2. weiblich  4. Niedersachsen/Bremen       0   
...            ...                      ...     ...   
70187  1. männlich   5. Nordrhein-Westfalen

In [80]:
#how many minutes per week each participant spent doing sports before the pandemic
extract["Sportstark_vorher"]=data["SCBX013_1"].astype(float)*60+data["SCBX013_2"].astype(float)
extract["Sportmittel_vorher"]=data["SCBX013_3"].astype(float)*60+data["SCBX013_4"].astype(float)
extract["Sportmild_vorher"]=data["SCBX013_5"].astype(float)*60+data["SCBX013_6"].astype(float)
#and since the pandemic
extract["Sportstark_nachher"]=data["SCBX014_1"].astype(float)*60+data["SCBX014_2"].astype(float)
extract["Sportmittel_nachher"]=data["SCBX014_3"].astype(float)*60+data["SCBX014_4"].astype(float)
extract["Sportmild_nachher"]=data["SCBX014_5"].astype(float)*60+data["SCBX014_6"].astype(float)
extract

za_nr             version                 doi        id_g    hhid_g  \
0      7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  10005901.0  100059.0   
1      7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  10005901.0  100059.0   
2      7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  10005901.0  100059.0   
3      7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  10005901.0  100059.0   
4      7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  10005901.0  100059.0   
...       ...                 ...                 ...         ...       ...   
70187  7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  49988201.0  499882.0   
70188  7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  49988201.0  499882.0   
70189  7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  49988201.0  499882.0   
70190  7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  49988201.0  499882.0   
70191  7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  49988201.0  499882.0   

       pid      dDatum              exp_group  Woche Teilnahme  ...  \
0      1.0  21.03.2020  2. Gruppe 2 (Samstag)    1.0     1. ja  ...   
1      1.0  28.03.2020  2. Gruppe 2 (Samstag)    2.0     1. ja  ...   
2      1.0  04.04.2020  2. Gruppe 2 (Samstag)    3.0     1. ja  ...   
3      1.0  11.04.2020  2. Gruppe 2 (Samstag)    4.0     1. ja  ...   
4      1.0  18.04.2020  2. Gruppe 2 (Samstag)    5.0     1. ja  ...   
...    ...         ...                    ...    ...       ...  ...   
70187  1.0  08.06.2020   4. Gruppe 4 (Montag)   12.0     1. ja  ...   
70188  1.0  15.06.2020   4. Gruppe 4 (Montag)   13.0     1. ja  ...   
70189  1.0  22.06.2020   4. Gruppe 4 (Montag)   14.0     1. ja  ...   
70190  1.0  29.06.2020   4. Gruppe 4 (Montag)   15.0     1. ja  ...   
70191  1.0  06.07.2020   4. Gruppe 4 (Montag)   16.0     1. ja  ...   

               state_residence partner               Treffen_vorher  \
0      4. Niedersachsen/Bremen          3. Mehrmals in dieser Woche   
1      4. Niedersachsen/Bremen          3. Mehrmals in dieser Woche   
2      4. Niedersachsen/Bremen          3. Mehrmals in dieser Woche   
3      4. Niedersachsen/Bremen          3. Mehrmals in dieser Woche   
4      4. Niedersachsen/Bremen       0  3. Mehrmals in dieser Woche   
...                        ...     ...                          ...   
70187   5. Nordrhein-Westfalen       1    2. Einmal in dieser Woche   
70188   5. Nordrhein-Westfalen            2. Einmal in dieser Woche   
70189   5. Nordrhein-Westfalen            2. Einmal in dieser Woche   
70190   5. Nordrhein-Westfalen       1    2. Einmal in dieser Woche   
70191   5. Nordrhein-Westfalen            2. Einmal in dieser Woche   

                   Treffen_nachher Sportstark_vorher Sportmittel_vorher  \
0                     1. Gar nicht               NaN                NaN   
1        2. Einmal in dieser Woche               NaN                NaN   
2                     1. Gar nicht             120.0              300.0   
3        2. Einmal in dieser Woche               NaN                NaN   
4      3. Mehrmals in dieser Woche               NaN                NaN   
...                            ...               ...                ...   
70187  3. Mehrmals in dieser Woche               NaN                NaN   
70188  3. Mehrmals in dieser Woche               NaN                NaN   
70189  3. Mehrmals in dieser Woche               NaN                NaN   
70190    2. Einmal in dieser Woche               NaN                NaN   
70191  3. Mehrmals in dieser Woche               NaN                NaN   

      Sportmild_vorher Sportstark_nachher Sportmittel_nachher  \
0                  NaN                NaN                 NaN   
1                  NaN                NaN                 NaN   
2                480.0              120.0               300.0   
3                  NaN                NaN                 NaN   
4                  NaN                NaN                 NaN   
...                ...                ...                 ... 

In [81]:
#how much time the participants spent with tv, computers etc. before and since the pandemic
extract["elektro_zeit_vorher"]=data["SCBX015_1"].astype(float)*60+data["SCBX015_2"].astype(float)
extract["elektro_zeit_nachher"]=data["SCBX016_1"].astype(float)*60+data["SCBX016_2"].astype(float)
extract

za_nr             version                 doi        id_g    hhid_g  \
0      7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  10005901.0  100059.0   
1      7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  10005901.0  100059.0   
2      7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  10005901.0  100059.0   
3      7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  10005901.0  100059.0   
4      7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  10005901.0  100059.0   
...       ...                 ...                 ...         ...       ...   
70187  7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  49988201.0  499882.0   
70188  7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  49988201.0  499882.0   
70189  7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  49988201.0  499882.0   
70190  7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  49988201.0  499882.0   
70191  7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  49988201.0  499882.0   

       pid      dDatum              exp_group  Woche Teilnahme  ...  \
0      1.0  21.03.2020  2. Gruppe 2 (Samstag)    1.0     1. ja  ...   
1      1.0  28.03.2020  2. Gruppe 2 (Samstag)    2.0     1. ja  ...   
2      1.0  04.04.2020  2. Gruppe 2 (Samstag)    3.0     1. ja  ...   
3      1.0  11.04.2020  2. Gruppe 2 (Samstag)    4.0     1. ja  ...   
4      1.0  18.04.2020  2. Gruppe 2 (Samstag)    5.0     1. ja  ...   
...    ...         ...                    ...    ...       ...  ...   
70187  1.0  08.06.2020   4. Gruppe 4 (Montag)   12.0     1. ja  ...   
70188  1.0  15.06.2020   4. Gruppe 4 (Montag)   13.0     1. ja  ...   
70189  1.0  22.06.2020   4. Gruppe 4 (Montag)   14.0     1. ja  ...   
70190  1.0  29.06.2020   4. Gruppe 4 (Montag)   15.0     1. ja  ...   
70191  1.0  06.07.2020   4. Gruppe 4 (Montag)   16.0     1. ja  ...   

                    Treffen_vorher              Treffen_nachher  \
0      3. Mehrmals in dieser Woche                 1. Gar nicht   
1      3. Mehrmals in dieser Woche    2. Einmal in dieser Woche   
2      3. Mehrmals in dieser Woche                 1. Gar nicht   
3      3. Mehrmals in dieser Woche    2. Einmal in dieser Woche   
4      3. Mehrmals in dieser Woche  3. Mehrmals in dieser Woche   
...                            ...                          ...   
70187    2. Einmal in dieser Woche  3. Mehrmals in dieser Woche   
70188    2. Einmal in dieser Woche  3. Mehrmals in dieser Woche   
70189    2. Einmal in dieser Woche  3. Mehrmals in dieser Woche   
70190    2. Einmal in dieser Woche    2. Einmal in dieser Woche   
70191    2. Einmal in dieser Woche  3. Mehrmals in dieser Woche   

      Sportstark_vorher Sportmittel_vorher Sportmild_vorher  \
0                   NaN                NaN              NaN   
1                   NaN                NaN              NaN   
2                 120.0              300.0            480.0   
3                   NaN                NaN              NaN   
4                   NaN                NaN              NaN   
...                 ...                ...              ...   
70187               NaN                NaN              NaN   
70188               NaN                NaN              NaN   
70189               NaN                NaN              NaN   
70190               NaN                NaN              NaN   
70191               NaN                NaN              NaN   

      Sportstark_nachher Sportmittel_nachher Sportmild_nachher  \
0                    NaN                 NaN               NaN   
1                    NaN                 NaN               NaN   
2                  120.0               300.0             480.0   
3                    NaN                 NaN               NaN   
4                    NaN                 NaN               NaN   
...                  ...                 ...               ...   
70187                NaN                 NaN               NaN   
70188                NaN                 NaN               NaN   
70189                NaN                 NaN      

In [82]:
#how many snacks before and since the pandemic per day
extract["Snacks_vorher"]=data["SCBX017"].astype(float)
extract["Snacks_nachher"]=data["SCBX018"].astype(float)
extract


za_nr             version                 doi        id_g    hhid_g  \
0      7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  10005901.0  100059.0   
1      7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  10005901.0  100059.0   
2      7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  10005901.0  100059.0   
3      7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  10005901.0  100059.0   
4      7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  10005901.0  100059.0   
...       ...                 ...                 ...         ...       ...   
70187  7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  49988201.0  499882.0   
70188  7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  49988201.0  499882.0   
70189  7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  49988201.0  499882.0   
70190  7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  49988201.0  499882.0   
70191  7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  49988201.0  499882.0   

       pid      dDatum              exp_group  Woche Teilnahme  ...  \
0      1.0  21.03.2020  2. Gruppe 2 (Samstag)    1.0     1. ja  ...   
1      1.0  28.03.2020  2. Gruppe 2 (Samstag)    2.0     1. ja  ...   
2      1.0  04.04.2020  2. Gruppe 2 (Samstag)    3.0     1. ja  ...   
3      1.0  11.04.2020  2. Gruppe 2 (Samstag)    4.0     1. ja  ...   
4      1.0  18.04.2020  2. Gruppe 2 (Samstag)    5.0     1. ja  ...   
...    ...         ...                    ...    ...       ...  ...   
70187  1.0  08.06.2020   4. Gruppe 4 (Montag)   12.0     1. ja  ...   
70188  1.0  15.06.2020   4. Gruppe 4 (Montag)   13.0     1. ja  ...   
70189  1.0  22.06.2020   4. Gruppe 4 (Montag)   14.0     1. ja  ...   
70190  1.0  29.06.2020   4. Gruppe 4 (Montag)   15.0     1. ja  ...   
70191  1.0  06.07.2020   4. Gruppe 4 (Montag)   16.0     1. ja  ...   

      Sportstark_vorher Sportmittel_vorher Sportmild_vorher  \
0                   NaN                NaN              NaN   
1                   NaN                NaN              NaN   
2                 120.0              300.0            480.0   
3                   NaN                NaN              NaN   
4                   NaN                NaN              NaN   
...                 ...                ...              ...   
70187               NaN                NaN              NaN   
70188               NaN                NaN              NaN   
70189               NaN                NaN              NaN   
70190               NaN                NaN              NaN   
70191               NaN                NaN              NaN   

      Sportstark_nachher Sportmittel_nachher Sportmild_nachher  \
0                    NaN                 NaN               NaN   
1                    NaN                 NaN               NaN   
2                  120.0               300.0             480.0   
3                    NaN                 NaN               NaN   
4                    NaN                 NaN               NaN   
...                  ...                 ...               ...   
70187                NaN                 NaN               NaN   
70188                NaN                 NaN               NaN   
70189                NaN                 NaN               NaN   
70190                NaN                 NaN               NaN   
70191              150.0                 0.0             360.0   

      elektro_zeit_vorher elektro_zeit_nachher Snacks_vorher Snacks_nachher  
0                     NaN                  NaN           NaN            NaN  
1                     NaN                  NaN           NaN            NaN  
2                     NaN                  NaN           NaN            NaN  
3                   240.0                420.0           2.0            2.0  
4                     NaN                  NaN           NaN            NaN  
...                   ...                  ...           ...            ...  
70187                 NaN                  NaN           NaN            NaN  
70188                 NaN       

In [83]:
#we get out certain moods since the pandemic
extract["aufgeregt_nachher"]=data["SCBX008"]
extract["besorgt_nachher"]=data["SCBX009"]
extract["beunruhigt_nachher"]=data["SCBX010"]
extract["nervoes_nachher"]=data["SCBX011"]

extract

za_nr             version                 doi        id_g    hhid_g  \
0      7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  10005901.0  100059.0   
1      7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  10005901.0  100059.0   
2      7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  10005901.0  100059.0   
3      7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  10005901.0  100059.0   
4      7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  10005901.0  100059.0   
...       ...                 ...                 ...         ...       ...   
70187  7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  49988201.0  499882.0   
70188  7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  49988201.0  499882.0   
70189  7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  49988201.0  499882.0   
70190  7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  49988201.0  499882.0   
70191  7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  49988201.0  499882.0   

       pid      dDatum              exp_group  Woche Teilnahme  ...  \
0      1.0  21.03.2020  2. Gruppe 2 (Samstag)    1.0     1. ja  ...   
1      1.0  28.03.2020  2. Gruppe 2 (Samstag)    2.0     1. ja  ...   
2      1.0  04.04.2020  2. Gruppe 2 (Samstag)    3.0     1. ja  ...   
3      1.0  11.04.2020  2. Gruppe 2 (Samstag)    4.0     1. ja  ...   
4      1.0  18.04.2020  2. Gruppe 2 (Samstag)    5.0     1. ja  ...   
...    ...         ...                    ...    ...       ...  ...   
70187  1.0  08.06.2020   4. Gruppe 4 (Montag)   12.0     1. ja  ...   
70188  1.0  15.06.2020   4. Gruppe 4 (Montag)   13.0     1. ja  ...   
70189  1.0  22.06.2020   4. Gruppe 4 (Montag)   14.0     1. ja  ...   
70190  1.0  29.06.2020   4. Gruppe 4 (Montag)   15.0     1. ja  ...   
70191  1.0  06.07.2020   4. Gruppe 4 (Montag)   16.0     1. ja  ...   

      Sportmittel_nachher Sportmild_nachher elektro_zeit_vorher  \
0                     NaN               NaN                 NaN   
1                     NaN               NaN                 NaN   
2                   300.0             480.0                 NaN   
3                     NaN               NaN               240.0   
4                     NaN               NaN                 NaN   
...                   ...               ...                 ...   
70187                 NaN               NaN                 NaN   
70188                 NaN               NaN                 NaN   
70189                 NaN               NaN                 NaN   
70190                 NaN               NaN                 NaN   
70191                 0.0             360.0                 NaN   

      elektro_zeit_nachher Snacks_vorher Snacks_nachher   aufgeregt_nachher  \
0                      NaN           NaN            NaN  1. überhaupt nicht   
1                      NaN           NaN            NaN  1. überhaupt nicht   
2                      NaN           NaN            NaN  1. überhaupt nicht   
3                    420.0           2.0            2.0  1. überhaupt nicht   
4                      NaN           NaN            NaN                 NaN   
...                    ...           ...            ...                 ...   
70187                  NaN           NaN            NaN                 NaN   
70188                  NaN           NaN            NaN                 NaN   
70189                300.0           NaN            7.0                 NaN   
70190                  NaN           NaN            NaN                 NaN   
70191                390.0           NaN            1.0                 NaN   

          besorgt_nachher  beunruhigt_nachher     nervoes_nachher  
0            2. ein wenig        2. ein wenig  1. überhaupt nicht  
1      1. überhaupt nicht        2. ein wenig  1. überhaupt nicht  
2            2. ein wenig  1. überhaupt nicht  1. überhaupt nicht  
3            2. ein wenig  1. überhaupt nicht  1. überhaupt nicht  
4      1. überhaupt nicht                 NaN  1. überhaupt nicht  
...                   ...                 ..

In [65]:
#how often participants felt this mood in the last 7 days
extract["deprimiert_nachher"]=data["SCBX019"]
extract["uninteressiert_nachher"]=data["SCBX020"]
extract["einsam_nachher"]=data["SCBX021"]
extract


za_nr             version                 doi        id_g    hhid_g  \
0      7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  10005901.0  100059.0   
1      7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  10005901.0  100059.0   
2      7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  10005901.0  100059.0   
3      7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  10005901.0  100059.0   
4      7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  10005901.0  100059.0   
...       ...                 ...                 ...         ...       ...   
70187  7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  49988201.0  499882.0   
70188  7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  49988201.0  499882.0   
70189  7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  49988201.0  499882.0   
70190  7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  49988201.0  499882.0   
70191  7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  49988201.0  499882.0   

       pid      dDatum              exp_group  Woche Teilnahme  ...  \
0      1.0  21.03.2020  2. Gruppe 2 (Samstag)    1.0     1. ja  ...   
1      1.0  28.03.2020  2. Gruppe 2 (Samstag)    2.0     1. ja  ...   
2      1.0  04.04.2020  2. Gruppe 2 (Samstag)    3.0     1. ja  ...   
3      1.0  11.04.2020  2. Gruppe 2 (Samstag)    4.0     1. ja  ...   
4      1.0  18.04.2020  2. Gruppe 2 (Samstag)    5.0     1. ja  ...   
...    ...         ...                    ...    ...       ...  ...   
70187  1.0  08.06.2020   4. Gruppe 4 (Montag)   12.0     1. ja  ...   
70188  1.0  15.06.2020   4. Gruppe 4 (Montag)   13.0     1. ja  ...   
70189  1.0  22.06.2020   4. Gruppe 4 (Montag)   14.0     1. ja  ...   
70190  1.0  29.06.2020   4. Gruppe 4 (Montag)   15.0     1. ja  ...   
70191  1.0  06.07.2020   4. Gruppe 4 (Montag)   16.0     1. ja  ...   

      elektro_zeit_nachher Snacks_vorher Snacks_nachher   aufgeregt_nachher  \
0                      NaN           NaN            NaN  1. überhaupt nicht   
1                      NaN           NaN            NaN  1. überhaupt nicht   
2                      NaN           NaN            NaN  1. überhaupt nicht   
3                    420.0           2.0            2.0  1. überhaupt nicht   
4                      NaN           NaN            NaN                 NaN   
...                    ...           ...            ...                 ...   
70187                  NaN           NaN            NaN                 NaN   
70188                  NaN           NaN            NaN                 NaN   
70189                300.0           NaN            7.0                 NaN   
70190                  NaN           NaN            NaN                 NaN   
70191                390.0           NaN            1.0                 NaN   

          besorgt_nachher  beunruhigt_nachher     nervoes_nachher  \
0            2. ein wenig        2. ein wenig  1. überhaupt nicht   
1      1. überhaupt nicht        2. ein wenig  1. überhaupt nicht   
2            2. ein wenig  1. überhaupt nicht  1. überhaupt nicht   
3            2. ein wenig  1. überhaupt nicht  1. überhaupt nicht   
4      1. überhaupt nicht                 NaN  1. überhaupt nicht   
...                   ...                 ...                 ...   
70187        2. ein wenig                 NaN  1. überhaupt nicht   
70188        2. ein wenig                 NaN  1. überhaupt nicht   
70189        2. ein wenig                 NaN  1. überhaupt nicht   
70190        2. ein wenig                 NaN  1. überhaupt nicht   
70191        2. ein wenig                 NaN  1. überhaupt nicht   

                              deprimiert_nachher  \
0                                            NaN   
1                                            NaN   
2                                            NaN   
3                                            NaN   
4      1. Überhaupt nicht oder weniger als 1 Tag   
...                                          ...   
70187  1. Überhaupt nicht oder weniger als 1 Tag   
70188  1. Üb

In [66]:
#how many hours on an average day would be used for work, including commute
extract["Beruf"]=data["SCBX022"]
#how many hours would be spent on an average day for care
extract["Care"]=data["SCBX023"]
extract

za_nr             version                 doi        id_g    hhid_g  \
0      7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  10005901.0  100059.0   
1      7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  10005901.0  100059.0   
2      7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  10005901.0  100059.0   
3      7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  10005901.0  100059.0   
4      7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  10005901.0  100059.0   
...       ...                 ...                 ...         ...       ...   
70187  7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  49988201.0  499882.0   
70188  7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  49988201.0  499882.0   
70189  7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  49988201.0  499882.0   
70190  7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  49988201.0  499882.0   
70191  7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  49988201.0  499882.0   

       pid      dDatum              exp_group  Woche Teilnahme  ...  \
0      1.0  21.03.2020  2. Gruppe 2 (Samstag)    1.0     1. ja  ...   
1      1.0  28.03.2020  2. Gruppe 2 (Samstag)    2.0     1. ja  ...   
2      1.0  04.04.2020  2. Gruppe 2 (Samstag)    3.0     1. ja  ...   
3      1.0  11.04.2020  2. Gruppe 2 (Samstag)    4.0     1. ja  ...   
4      1.0  18.04.2020  2. Gruppe 2 (Samstag)    5.0     1. ja  ...   
...    ...         ...                    ...    ...       ...  ...   
70187  1.0  08.06.2020   4. Gruppe 4 (Montag)   12.0     1. ja  ...   
70188  1.0  15.06.2020   4. Gruppe 4 (Montag)   13.0     1. ja  ...   
70189  1.0  22.06.2020   4. Gruppe 4 (Montag)   14.0     1. ja  ...   
70190  1.0  29.06.2020   4. Gruppe 4 (Montag)   15.0     1. ja  ...   
70191  1.0  06.07.2020   4. Gruppe 4 (Montag)   16.0     1. ja  ...   

      Snacks_nachher   aufgeregt_nachher     besorgt_nachher  \
0                NaN  1. überhaupt nicht        2. ein wenig   
1                NaN  1. überhaupt nicht  1. überhaupt nicht   
2                NaN  1. überhaupt nicht        2. ein wenig   
3                2.0  1. überhaupt nicht        2. ein wenig   
4                NaN                 NaN  1. überhaupt nicht   
...              ...                 ...                 ...   
70187            NaN                 NaN        2. ein wenig   
70188            NaN                 NaN        2. ein wenig   
70189            7.0                 NaN        2. ein wenig   
70190            NaN                 NaN        2. ein wenig   
70191            1.0                 NaN        2. ein wenig   

       beunruhigt_nachher     nervoes_nachher  \
0            2. ein wenig  1. überhaupt nicht   
1            2. ein wenig  1. überhaupt nicht   
2      1. überhaupt nicht  1. überhaupt nicht   
3      1. überhaupt nicht  1. überhaupt nicht   
4                     NaN  1. überhaupt nicht   
...                   ...                 ...   
70187                 NaN  1. überhaupt nicht   
70188                 NaN  1. überhaupt nicht   
70189                 NaN  1. überhaupt nicht   
70190                 NaN  1. überhaupt nicht   
70191                 NaN  1. überhaupt nicht   

                              deprimiert_nachher  \
0                                            NaN   
1                                            NaN   
2                                            NaN   
3                                            NaN   
4      1. Überhaupt nicht oder weniger als 1 Tag   
...                                          ...   
70187  1. Überhaupt nicht oder weniger als 1 Tag   
70188  1. Überhaupt nicht oder weniger als 1 Tag   
70189  1. Überhaupt nicht oder weniger als 1 Tag   
70190  1. Überhaupt nicht oder weniger als 1 Tag   
70191  1. Überhaupt nicht oder weniger als 1 Tag   

                          uninteressiert_nachher  \
0                                            NaN   
1                                            NaN   
2                                            NaN   
3              

In [67]:
#we convert the rating of perceived contentness with work and family
extract["Arbeit_Rating_nachher"]=data["SCBX024"].str[:2]
extract["Arbeit_Rating_nachher"]=extract["Arbeit_Rating_nachher"].str.replace(".","")
extract["Arbeit_Rating_nachher"]=extract["Arbeit_Rating_nachher"].astype(float)
extract["Familie_Rating_nachher"]=data["SCBX025"].str[:2]
extract["Familie_Rating_nachher"]=extract["Familie_Rating_nachher"].str.replace(".","")
extract["Familie_Rating_nachher"]=extract["Familie_Rating_nachher"].astype(float)
extract

za_nr             version                 doi        id_g    hhid_g  \
0      7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  10005901.0  100059.0   
1      7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  10005901.0  100059.0   
2      7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  10005901.0  100059.0   
3      7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  10005901.0  100059.0   
4      7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  10005901.0  100059.0   
...       ...                 ...                 ...         ...       ...   
70187  7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  49988201.0  499882.0   
70188  7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  49988201.0  499882.0   
70189  7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  49988201.0  499882.0   
70190  7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  49988201.0  499882.0   
70191  7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  49988201.0  499882.0   

       pid      dDatum              exp_group  Woche Teilnahme  ...  \
0      1.0  21.03.2020  2. Gruppe 2 (Samstag)    1.0     1. ja  ...   
1      1.0  28.03.2020  2. Gruppe 2 (Samstag)    2.0     1. ja  ...   
2      1.0  04.04.2020  2. Gruppe 2 (Samstag)    3.0     1. ja  ...   
3      1.0  11.04.2020  2. Gruppe 2 (Samstag)    4.0     1. ja  ...   
4      1.0  18.04.2020  2. Gruppe 2 (Samstag)    5.0     1. ja  ...   
...    ...         ...                    ...    ...       ...  ...   
70187  1.0  08.06.2020   4. Gruppe 4 (Montag)   12.0     1. ja  ...   
70188  1.0  15.06.2020   4. Gruppe 4 (Montag)   13.0     1. ja  ...   
70189  1.0  22.06.2020   4. Gruppe 4 (Montag)   14.0     1. ja  ...   
70190  1.0  29.06.2020   4. Gruppe 4 (Montag)   15.0     1. ja  ...   
70191  1.0  06.07.2020   4. Gruppe 4 (Montag)   16.0     1. ja  ...   

          besorgt_nachher  beunruhigt_nachher     nervoes_nachher  \
0            2. ein wenig        2. ein wenig  1. überhaupt nicht   
1      1. überhaupt nicht        2. ein wenig  1. überhaupt nicht   
2            2. ein wenig  1. überhaupt nicht  1. überhaupt nicht   
3            2. ein wenig  1. überhaupt nicht  1. überhaupt nicht   
4      1. überhaupt nicht                 NaN  1. überhaupt nicht   
...                   ...                 ...                 ...   
70187        2. ein wenig                 NaN  1. überhaupt nicht   
70188        2. ein wenig                 NaN  1. überhaupt nicht   
70189        2. ein wenig                 NaN  1. überhaupt nicht   
70190        2. ein wenig                 NaN  1. überhaupt nicht   
70191        2. ein wenig                 NaN  1. überhaupt nicht   

                              deprimiert_nachher  \
0                                            NaN   
1                                            NaN   
2                                            NaN   
3                                            NaN   
4      1. Überhaupt nicht oder weniger als 1 Tag   
...                                          ...   
70187  1. Überhaupt nicht oder weniger als 1 Tag   
70188  1. Überhaupt nicht oder weniger als 1 Tag   
70189  1. Überhaupt nicht oder weniger als 1 Tag   
70190  1. Überhaupt nicht oder weniger als 1 Tag   
70191  1. Überhaupt nicht oder weniger als 1 Tag   

                          uninteressiert_nachher  \
0                                            NaN   
1                                            NaN   
2                                            NaN   
3                                            NaN   
4      1. Überhaupt nicht oder weniger als 1 Tag   
...                                          ...   
70187  1. Überhaupt nicht oder weniger als 1 Tag   
70188  1. Überhaupt nicht oder weniger als 1 Tag   
70189                            2. An 1-2 Tagen   
70190  1. Überhaupt nicht oder weniger als 1 Tag   
70191  1. Überhaupt nicht oder weniger als 1 Tag   

                                  einsam_nachher Beruf  Care  \
0                                            NaN   NaN   NaN   

In [68]:
#we also check for how many kids are in the household
extract["Kinder_nachher"]=data["SCFX001"].str[:1]
extract["Kinder_nachher"]=extract["Kinder_nachher"].str.replace("-","")
extract["Kinder_nachher"]=pd.to_numeric(extract["Kinder_nachher"])
extract["Kinder_nachher"].unique()
data["SCFX002_a"].unique()

[NaN, '0. item not checked', '1. item checked']
Categories (2, object): ['0. item not checked' < '1. item checked']

In [69]:
#we include how kids were taken care of before the pandemic  
extract["kita_vorher"]=data["SCFX002_a"].str[:1]
extract["grundschule_vorher"]=data["SCFX002_b"].str[:1]
extract["weiterfuehrendeschule_vorher"]=data["SCFX002_c"].str[:1]
extract["anderePerson60_vorher"]=data["SCFX002_d"].str[:1]
extract["keinePerson_vorher"]=data["SCFX002_e"].str[:1]

extract

za_nr             version                 doi        id_g    hhid_g  \
0      7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  10005901.0  100059.0   
1      7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  10005901.0  100059.0   
2      7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  10005901.0  100059.0   
3      7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  10005901.0  100059.0   
4      7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  10005901.0  100059.0   
...       ...                 ...                 ...         ...       ...   
70187  7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  49988201.0  499882.0   
70188  7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  49988201.0  499882.0   
70189  7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  49988201.0  499882.0   
70190  7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  49988201.0  499882.0   
70191  7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  49988201.0  499882.0   

       pid      dDatum              exp_group  Woche Teilnahme  ... Beruf  \
0      1.0  21.03.2020  2. Gruppe 2 (Samstag)    1.0     1. ja  ...   NaN   
1      1.0  28.03.2020  2. Gruppe 2 (Samstag)    2.0     1. ja  ...   NaN   
2      1.0  04.04.2020  2. Gruppe 2 (Samstag)    3.0     1. ja  ...   NaN   
3      1.0  11.04.2020  2. Gruppe 2 (Samstag)    4.0     1. ja  ...   NaN   
4      1.0  18.04.2020  2. Gruppe 2 (Samstag)    5.0     1. ja  ...   8.0   
...    ...         ...                    ...    ...       ...  ...   ...   
70187  1.0  08.06.2020   4. Gruppe 4 (Montag)   12.0     1. ja  ...   8.0   
70188  1.0  15.06.2020   4. Gruppe 4 (Montag)   13.0     1. ja  ...   8.0   
70189  1.0  22.06.2020   4. Gruppe 4 (Montag)   14.0     1. ja  ...   8.0   
70190  1.0  29.06.2020   4. Gruppe 4 (Montag)   15.0     1. ja  ...   8.0   
70191  1.0  06.07.2020   4. Gruppe 4 (Montag)   16.0     1. ja  ...   0.0   

       Care Arbeit_Rating_nachher Familie_Rating_nachher Kinder_nachher  \
0       NaN                   NaN                    NaN            0.0   
1       NaN                   NaN                    NaN            0.0   
2       NaN                   NaN                    NaN            0.0   
3       NaN                   NaN                    NaN            0.0   
4       2.0                   7.0                    5.0            0.0   
...     ...                   ...                    ...            ...   
70187   0.0                   5.0                    9.0            0.0   
70188   0.0                   9.0                    9.0            0.0   
70189   0.0                   8.0                    8.0            0.0   
70190   0.0                   NaN                    9.0            1.0   
70191  24.0                   NaN                    8.0            1.0   

      kita_vorher grundschule_vorher weiterfuehrendeschule_vorher  \
0             NaN                NaN                          NaN   
1             NaN                NaN                          NaN   
2             NaN                NaN                          NaN   
3             NaN                NaN                          NaN   
4             NaN                NaN                          NaN   
...           ...                ...                          ...   
70187         NaN                NaN                          NaN   
70188         NaN                NaN                          NaN   
70189         NaN                NaN                          NaN   
70190         NaN                NaN                          NaN   
70191         NaN                NaN                          NaN   

      anderePerson60_vorher keinePerson_vorher  
0                       NaN                NaN  
1                       NaN                NaN  
2                       NaN                NaN  
3                       NaN                NaN  
4                       NaN                NaN  
...                     ...                ...  
70187                   NaN                NaN  
70188       

In [73]:
#we include how kids were taken care of since the pandemic  
extract["selber_nachher"]=data["SCFX003_a"].str[:1]
extract["notfallkita_nachher"]=data["SCFX003_b"].str[:1]
extract["anderePersonHaushalt_nachher"]=data["SCFX003_c"].str[:1]
extract["fremderHaushalt_nachher"]=data["SCFX003_d"].str[:1]
extract["anderePersonHaushalt60_nachher"]=data["SCFX003_e"].str[:1]
extract["fremderHaushalt60_nachher"]=data["SCFX003_f"].str[:1]
extract["nichtbetreut_nachher"]=data["SCFX003_g"].str[:1]
extract["kita_nachher"]=data["SCFX003_h"].str[:1]
extract

za_nr             version                 doi        id_g    hhid_g  \
0      7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  10005901.0  100059.0   
1      7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  10005901.0  100059.0   
2      7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  10005901.0  100059.0   
3      7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  10005901.0  100059.0   
4      7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  10005901.0  100059.0   
...       ...                 ...                 ...         ...       ...   
70187  7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  49988201.0  499882.0   
70188  7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  49988201.0  499882.0   
70189  7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  49988201.0  499882.0   
70190  7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  49988201.0  499882.0   
70191  7745.0  1.0.0 (2021-02-23)  doi:10.4232/1.3700  49988201.0  499882.0   

       pid      dDatum              exp_group  Woche Teilnahme  ...  \
0      1.0  21.03.2020  2. Gruppe 2 (Samstag)    1.0     1. ja  ...   
1      1.0  28.03.2020  2. Gruppe 2 (Samstag)    2.0     1. ja  ...   
2      1.0  04.04.2020  2. Gruppe 2 (Samstag)    3.0     1. ja  ...   
3      1.0  11.04.2020  2. Gruppe 2 (Samstag)    4.0     1. ja  ...   
4      1.0  18.04.2020  2. Gruppe 2 (Samstag)    5.0     1. ja  ...   
...    ...         ...                    ...    ...       ...  ...   
70187  1.0  08.06.2020   4. Gruppe 4 (Montag)   12.0     1. ja  ...   
70188  1.0  15.06.2020   4. Gruppe 4 (Montag)   13.0     1. ja  ...   
70189  1.0  22.06.2020   4. Gruppe 4 (Montag)   14.0     1. ja  ...   
70190  1.0  29.06.2020   4. Gruppe 4 (Montag)   15.0     1. ja  ...   
70191  1.0  06.07.2020   4. Gruppe 4 (Montag)   16.0     1. ja  ...   

      keinePerson_vorher selber_nachher notfallkita_nachher  \
0                    NaN            NaN                 NaN   
1                    NaN            NaN                 NaN   
2                    NaN            NaN                 NaN   
3                    NaN            NaN                 NaN   
4                    NaN            NaN                 NaN   
...                  ...            ...                 ...   
70187                NaN            NaN                 NaN   
70188                NaN            NaN                 NaN   
70189                NaN            NaN                 NaN   
70190                NaN              1                 NaN   
70191                NaN              1                 NaN   

      anderePersonHaushalt_nachher fremderHaushalt_nachher  \
0                              NaN                     NaN   
1                              NaN                     NaN   
2                              NaN                     NaN   
3                              NaN                     NaN   
4                              NaN                     NaN   
...                            ...                     ...   
70187                          NaN                     NaN   
70188                          NaN                     NaN   
70189                          NaN                     NaN   
70190                            0                       0   
70191                            0                       0   

      anderePersonHaushalt60_nachher fremderHaushalt60_nachher  \
0                                NaN                       NaN   
1                                NaN                       NaN   
2                                NaN                       NaN   
3                                NaN                       NaN   
4                                NaN                       NaN   
...                              ...                       ...   
70187                            NaN                       NaN   
70188                            NaN                       NaN   
70189                            NaN                       NaN   
70190                              0        

In [71]:
#we create one dataframe to just check out before and after pandemic childcare
basedf=pd.DataFrame()
#so we iterate and check for each form of childcare (a to e) whether the item is checked
basedf["childcare_vorher"]="0"
for buchstabe in ["a","b","c","d","e"]:
    print("SCFX002_"+str(buchstabe))
    currentdf=pd.DataFrame()
    #and write out id and week
    currentdf["id"]=data[data["SCFX002_"+buchstabe]=='1. item checked']["id_g"].astype(str).copy()
    currentdf["woche"]=data[data["SCFX002_"+buchstabe]=='1. item checked']["Woche"].astype(str).copy()
    #plus the letter of this type of childcare
    currentdf["childcare_vorher"]=str(buchstabe)
    #and concat
    basedf=pd.concat([basedf,currentdf])
#we drop duplicates
basedf=basedf.drop_duplicates()

#then we do the same for types of childcare since pandemic 
basedf_two=pd.DataFrame()
basedf_two["childcare_nachher"]="0"
for buchstabe in ["a","b","c","d","e","f","g","h"]:
    print("SCFX003_"+str(buchstabe))
    currentdf=pd.DataFrame()
    currentdf["id"]=data[data["SCFX003_"+buchstabe]=='1. item checked']["id_g"].astype(str).copy()
    currentdf["woche"]=data[data["SCFX003_"+buchstabe]=='1. item checked']["Woche"].astype(str).copy()

    currentdf["childcare_nachher"]=str(buchstabe)
    basedf_two=pd.concat([basedf_two,currentdf])
basedf_two=basedf_two.drop_duplicates()
basedf_two
#then we merge this by using id and week
childcare_df=pd.merge(basedf,basedf_two,on=["id","woche"])
childcare_df.id=childcare_df.id.astype(float)
childcare_df.woche=childcare_df.woche.astype(float)
childcare_df

SCFX002_a
SCFX002_b
SCFX002_c
SCFX002_d
SCFX002_e
SCFX003_a
SCFX003_b
SCFX003_c
SCFX003_d
SCFX003_e
SCFX003_f
SCFX003_g
SCFX003_h


childcare_vorher          id  woche childcare_nachher
0                    a  10943001.0    1.0                 a
1                    a  10943001.0    1.0                 c
2                    a  10943001.0    2.0                 a
3                    a  10943001.0    7.0                 a
4                    a  10943001.0    8.0                 a
...                ...         ...    ...               ...
22315                e  49932501.0   14.0                 c
22316                e  49932501.0   15.0                 a
22317                e  49932501.0   15.0                 c
22318                e  49932501.0   16.0                 a
22319                e  49932501.0   16.0                 c

[22320 rows x 4 columns]

In [19]:
#also we create two lists of ids based on whether someone has a partner or not
singlelist=extract[(extract["partner"]=="0")]["id_g"].unique().tolist()
partnerlist=extract[(extract["partner"]=="1")]["id_g"].unique().tolist()


In [20]:
#then we create a dictionary of datasubsets based on whether parents are single or partnered and whether they are male or female
df_dict={}
df_dict["single_mothers"]=extract[(extract["gender_19"]=="2. weiblich")&(extract["Kinder_nachher"]>0)&(extract["id_g"].isin(singlelist))]
df_dict["single_fathers"]=extract[(extract["gender_19"]=="1. männlich")&(extract["Kinder_nachher"]>0)&(extract["id_g"].isin(singlelist))]
df_dict["partner_mothers"]=extract[(extract["gender_19"]=="2. weiblich")&(extract["Kinder_nachher"]>0)&(extract["id_g"].isin(partnerlist))]
df_dict["partner_fathers"]=extract[(extract["gender_19"]=="1. männlich")&(extract["Kinder_nachher"]>0)&(extract["id_g"].isin(partnerlist))]

In [21]:
print(len(df_dict["single_mothers"]["id_g"].unique()))
len(df_dict["single_fathers"]["id_g"].unique())

69


33

In [22]:
print(len(df_dict["partner_mothers"]["id_g"].unique()))
print(len(df_dict["partner_fathers"]["id_g"].unique()))

477
469


In [23]:
#we build a list of metrics that we have values for before the pandemic and that are numeric
print(df_dict["single_mothers"].columns[df_dict["single_mothers"].columns.str.contains("_vorher")])
metriclist=df_dict["single_mothers"].columns[df_dict["single_mothers"].columns.str.contains("_vorher")][1:6]

metriclist

Index(['Treffen_vorher', 'Sportstark_vorher', 'Sportmittel_vorher',
       'Sportmild_vorher', 'elektro_zeit_vorher', 'Snacks_vorher',
       'kita_vorher', 'grundschule_vorher', 'weiterfuehrendeschule_vorher',
       'anderePerson60_vorher', 'keinePerson_vorher'],
      dtype='object')


Index(['Sportstark_vorher', 'Sportmittel_vorher', 'Sportmild_vorher',
       'elektro_zeit_vorher', 'Snacks_vorher'],
      dtype='object')

In [24]:
#now we want a dataframe where we can trace back how the values have changed from before the pandemic
concat_metrics_subset=pd.DataFrame()

#we iterate through these metrics
for vorher in metriclist:
    nachher=(vorher.replace("_vorher","_nachher"))
    print(nachher)
    concat_subset=pd.DataFrame()
    #we iterate through the different subsets
    for key in df_dict.keys():
        #we write into a dictionary that will store the mean values
        dict_of_values={}
        #and we iterate through the weeks from 1 to 16
        for i in range(16):
            #if we have for this metric and for this subset not a nan for mean
            if math.isnan(df_dict[key][df_dict[key]["Woche"]==i][nachher].mean())==False:
                #we write this to our dictionary of values
                dict_of_values[i]=(df_dict[key][df_dict[key]["Woche"]==i][nachher].mean())
        #we also write the mean value for before the pandemic
        dict_of_values["vorher"]=(pd.to_numeric(df_dict[key][vorher]).mean())
        #and then we create a dataframe from this dictionary
        df=pd.DataFrame(dict_of_values, index=[0])
        #and write into a column which subset this is
        df["subset"]=key
        #and then concat this df to other dfs of other subsets
        concat_subset=pd.concat([concat_subset,df])
    #and once we are done through with all subsets
    concat_subset["metric"]=nachher.replace("_nachher","")
    #we add this df for one metric and all subsets to the greater df
    concat_metrics_subset=pd.concat([concat_metrics_subset,concat_subset])
concat_metrics_subset

Sportstark_nachher
Sportmittel_nachher
Sportmild_nachher
elektro_zeit_nachher
Snacks_nachher


3           9          11          14      vorher  \
0   84.824561  128.921569   76.577778   90.296296   94.508772   
0   73.571429  197.666667  151.952381  218.750000  133.142857   
0   66.312821   65.710875   87.680597   92.394958   73.466667   
0   95.161376  134.940217  152.186186  181.586716  113.769841   
0  325.912281  425.137255  324.522727  492.333333  299.245614   
0  247.190476  230.523810  215.333333  257.692308  166.761905   
0  190.135897  275.249337  293.687688  272.628692  219.151282   
0  203.089947  294.391304  334.146341  356.114391  216.322751   
0  676.087719  532.411765  534.369565  579.370370  814.824561   
0  350.523810  414.095238  342.761905  580.461538  256.761905   
0  376.751282  428.928382  455.749254  454.601660  382.507692   
0  349.243386  447.176630  496.213213  498.434944  358.650794   
0         NaN  346.607843  343.595745  336.083333  269.207547   
0         NaN  423.000000  361.000000  318.571429  315.727273   
0         NaN  331.915119  294.074074  301.886121  253.098667   
0         NaN  402.230978  345.910663  365.996764  350.701847   
0         NaN    1.823529    1.724490    2.677083    1.556604   
0         NaN    1.952381    1.509091    2.315789    1.409091   
0         NaN    1.835809    1.803581    1.895251    1.649867   
0         NaN    1.782065    1.612291    1.944134    1.592348   

            subset        metric           4  
0   single_mothers    Sportstark         NaN  
0   single_fathers    Sportstark         NaN  
0  partner_mothers    Sportstark         NaN  
0  partner_fathers    Sportstark         NaN  
0   single_mothers   Sportmittel         NaN  
0   single_fathers   Sportmittel         NaN  
0  partner_mothers   Sportmittel         NaN  
0  partner_fathers   Sportmittel         NaN  
0   single_mothers     Sportmild         NaN  
0   single_fathers     Sportmild         NaN  
0  partner_mothers     Sportmild         NaN  
0  partner_fathers     Sportmild         NaN  
0   single_mothers  elektro_zeit  361.566038  
0   single_fathers  elektro_zeit  350.272727  
0  partner_mothers  elektro_zeit  298.904000  
0  partner_fathers  elektro_zeit  372.052770  
0   single_mothers        Snacks    2.481132  
0   single_fathers        Snacks    2.136364  
0  partner_mothers        Snacks    2.343200  
0  partner_fathers        Snacks    2.094195

In [25]:
concat_metrics_subset[["subset","metric","vorher",14]].sort_values(["subset","metric"])

subset        metric      vorher          14
0  partner_fathers        Snacks    1.592348    1.944134
0  partner_fathers     Sportmild  358.650794  498.434944
0  partner_fathers   Sportmittel  216.322751  356.114391
0  partner_fathers    Sportstark  113.769841  181.586716
0  partner_fathers  elektro_zeit  350.701847  365.996764
0  partner_mothers        Snacks    1.649867    1.895251
0  partner_mothers     Sportmild  382.507692  454.601660
0  partner_mothers   Sportmittel  219.151282  272.628692
0  partner_mothers    Sportstark   73.466667   92.394958
0  partner_mothers  elektro_zeit  253.098667  301.886121
0   single_fathers        Snacks    1.409091    2.315789
0   single_fathers     Sportmild  256.761905  580.461538
0   single_fathers   Sportmittel  166.761905  257.692308
0   single_fathers    Sportstark  133.142857  218.750000
0   single_fathers  elektro_zeit  315.727273  318.571429
0   single_mothers        Snacks    1.556604    2.677083
0   single_mothers     Sportmild  814.824561  579.370370
0   single_mothers   Sportmittel  299.245614  492.333333
0   single_mothers    Sportstark   94.508772   90.296296
0   single_mothers  elektro_zeit  269.207547  336.083333

## Results, Part1
* So we see that actually fathers and mothers in partnerships have done  more excercise
* Single mothers have done less
* the consume of TV etc. has increased for women especially for single mothers

In [26]:
extract.columns

Index(['za_nr', 'version', 'doi', 'id_g', 'hhid_g', 'pid', 'dDatum',
       'exp_group', 'Woche', 'Teilnahme', 'sample', 'educ_school_19',
       'educ_job_19', 'marital_status_19', 'number_hh_members_19',
       'german_citizenship_19', 'internet_usage_19', 'state_19',
       'occupation_19', 'year_of_birth_cat_19', 'gender_19', 'state_residence',
       'partner', 'Treffen_vorher', 'Treffen_nachher', 'Sportstark_vorher',
       'Sportmittel_vorher', 'Sportmild_vorher', 'Sportstark_nachher',
       'Sportmittel_nachher', 'Sportmild_nachher', 'elektro_zeit_vorher',
       'elektro_zeit_nachher', 'Snacks_vorher', 'Snacks_nachher',
       'aufgeregt_nachher', 'besorgt_nachher', 'beunruhigt_nachher',
       'nervoes_nachher', 'deprimiert_nachher', 'uninteressiert_nachher',
       'einsam_nachher', 'Beruf', 'Care', 'Arbeit_Rating_nachher',
       'Familie_Rating_nachher', 'Kinder_nachher', 'kita_vorher',
       'grundschule_vorher', 'weiterfuehrendeschule_vorher',
       'anderePerson60_vo

In [27]:
#we also check how the social contacts have changed for each subgroup
subgroup="partner_fathers"
df_dict[subgroup][df_dict[subgroup]["Woche"]==16]["Treffen_vorher"].value_counts(normalize=True),df_dict[subgroup]["Treffen_vorher"].value_counts(normalize=True)




(3. Mehrmals in dieser Woche        0.383041
 2. Einmal in dieser Woche          0.374269
 1. Gar nicht                       0.128655
 4. Täglich oder mehrmals am Tag    0.099415
 -99. weiß nicht                    0.014620
 Name: Treffen_vorher, dtype: float64,
 3. Mehrmals in dieser Woche        0.385219
 2. Einmal in dieser Woche          0.370029
 1. Gar nicht                       0.125960
 4. Täglich oder mehrmals am Tag    0.108210
 -99. weiß nicht                    0.010582
 Name: Treffen_vorher, dtype: float64)

In [51]:
df_dict["partner_mothers"]["Treffen_nachher"].value_counts(normalize=True).sort_index(),df_dict["partner_mothers"]["Treffen_vorher"].value_counts(normalize=True).sort_index()

(-99. weiß nicht                    0.003028
 1. Gar nicht                       0.343650
 2. Einmal in dieser Woche          0.346005
 3. Mehrmals in dieser Woche        0.281918
 4. Täglich oder mehrmals am Tag    0.025399
 Name: Treffen_nachher, dtype: float64,
 -99. weiß nicht                    0.011224
 1. Gar nicht                       0.119048
 2. Einmal in dieser Woche          0.249150
 3. Mehrmals in dieser Woche        0.505442
 4. Täglich oder mehrmals am Tag    0.115136
 Name: Treffen_vorher, dtype: float64)

In [52]:
df_dict["single_mothers"]["Treffen_nachher"].value_counts(normalize=True).sort_index(),df_dict["single_mothers"]["Treffen_vorher"].value_counts(normalize=True).sort_index()

(-99. weiß nicht                    0.002451
 1. Gar nicht                       0.276961
 2. Einmal in dieser Woche          0.329657
 3. Mehrmals in dieser Woche        0.333333
 4. Täglich oder mehrmals am Tag    0.057598
 Name: Treffen_nachher, dtype: float64,
 -99. weiß nicht                    0.020910
 1. Gar nicht                       0.082411
 2. Einmal in dieser Woche          0.221402
 3. Mehrmals in dieser Woche        0.522755
 4. Täglich oder mehrmals am Tag    0.152522
 Name: Treffen_vorher, dtype: float64)

In [53]:
df_dict["partner_fathers"]["Treffen_nachher"].value_counts(normalize=True).sort_index(),df_dict["partner_fathers"]["Treffen_vorher"].value_counts(normalize=True).sort_index()

(-99. weiß nicht                    0.000679
 1. Gar nicht                       0.342043
 2. Einmal in dieser Woche          0.381235
 3. Mehrmals in dieser Woche        0.247201
 4. Täglich oder mehrmals am Tag    0.028843
 Name: Treffen_nachher, dtype: float64,
 -99. weiß nicht                    0.010582
 1. Gar nicht                       0.125960
 2. Einmal in dieser Woche          0.370029
 3. Mehrmals in dieser Woche        0.385219
 4. Täglich oder mehrmals am Tag    0.108210
 Name: Treffen_vorher, dtype: float64)

In [31]:
for key in df_dict.keys():
    print(key)
    #df_dict[subgroup][df_dict[key]["Woche"]==16]
    print(df_dict[key][df_dict[key]["Woche"]==16]["Treffen_nachher"].value_counts(normalize=True)[["1. Gar nicht","2. Einmal in dieser Woche"]].sum())
    print(df_dict[key][df_dict[key]["Woche"]==16]["Treffen_vorher"].value_counts(normalize=True)[["1. Gar nicht","2. Einmal in dieser Woche"]].sum())

single_mothers
0.2857142857142857
0.23809523809523808
single_fathers
0.2857142857142857
0.3333333333333333
partner_mothers
0.45808383233532934
0.3567073170731707
partner_fathers
0.527536231884058
0.5029239766081871


In [32]:
for key in df_dict.keys():
    print(key)
    print(df_dict[key]["Treffen_nachher"].value_counts(normalize=True)[["1. Gar nicht","2. Einmal in dieser Woche"]].sum())
    print(df_dict[key]["Treffen_vorher"].value_counts(normalize=True)[["1. Gar nicht","2. Einmal in dieser Woche"]].sum())

single_mothers
0.6066176470588236
0.3038130381303813
single_fathers
0.6340057636887608
0.3343023255813954
partner_mothers
0.6896551724137931
0.3681972789115646
partner_fathers
0.7232779097387173
0.4959890766342379


## Results, Part2
* Mothers have suffered more, as they lost most of the contact to others
* Partnered mothers decrease was the biggest, single mothers the lowest value
* However, overall the absolute share of people with only one contact or less than that was the biggest for partnered fathers

In [33]:
#and we check how the number of work and care hours per day have changed through the pandemic
for i in extract.Woche.unique():
    print(pd.to_numeric(df_dict["partner_mothers"][df_dict["partner_mothers"].Woche==i]["Beruf"]).mean())
    print(pd.to_numeric(df_dict["partner_mothers"][df_dict["partner_mothers"].Woche==i]["Care"]).mean())

nan
nan
nan
nan
nan
nan
nan
nan
4.097989949748744
11.032663316582914
nan
nan
nan
nan
nan
nan
nan
nan
4.225806451612903
10.010666666666667
4.795580110497237
10.113259668508288
4.313186813186813
10.311475409836065
4.532763532763533
9.745042492917847
4.578212290502793
9.794444444444444
4.663841807909605
9.625352112676056
4.784848484848485
9.510574018126889


In [34]:
for i in extract.Woche.unique():
    print(pd.to_numeric(df_dict["single_mothers"][df_dict["single_mothers"].Woche==i]["Beruf"]).mean())
    print(pd.to_numeric(df_dict["single_mothers"][df_dict["single_mothers"].Woche==i]["Care"]).mean())

nan
nan
nan
nan
nan
nan
nan
nan
4.981132075471698
10.377358490566039
nan
nan
nan
nan
nan
nan
nan
nan
6.137254901960785
9.28
5.791666666666667
8.270833333333334
5.1020408163265305
8.708333333333334
5.296296296296297
8.074074074074074
5.5
8.5
5.456521739130435
7.586956521739131
5.195121951219512
7.195121951219512


In [35]:
for i in extract.Woche.unique():
    print(pd.to_numeric(df_dict["partner_fathers"][df_dict["partner_fathers"].Woche==i]["Beruf"]).mean())
    print(pd.to_numeric(df_dict["partner_fathers"][df_dict["partner_fathers"].Woche==i]["Care"]).mean())

nan
nan
nan
nan
nan
nan
nan
nan
6.8040201005025125
5.9874371859296485
nan
nan
nan
nan
nan
nan
nan
nan
7.650406504065041
4.937669376693767
7.873949579831932
4.92436974789916
7.512676056338028
5.070422535211268
7.6525423728813555
4.8668555240793205
7.818941504178273
4.506963788300836
7.791549295774648
4.326760563380281
7.2
4.855072463768116


## Results, Part3
* For all constellations that the amount of care work has decreased over the course of the pandemic for all groups
* However, especially for partnered mothers the care work has become a fulltime job (9h)

In [36]:
#then we also want to check how the form of care has changed by using the childcare_df and filtering it 
#through a merge with the df that only holds single mothers, partnered mothers, partnered fathers
single_mothers_care=pd.merge(df_dict["single_mothers"],childcare_df,left_on=["id_g","Woche"],right_on=["id","woche"])
partner_mothers_care=pd.merge(df_dict["partner_mothers"],childcare_df,left_on=["id_g","Woche"],right_on=["id","woche"])
partner_fathers_care=pd.merge(df_dict["partner_fathers"],childcare_df,left_on=["id_g","Woche"],right_on=["id","woche"])


*Vorher*

Kita / Kindergarten	SCFX002_a

Grundschule	SCFX002_b

Weiterführende Schule	SCFX002_c

Eine Person im Alter von 60 Jahren oder älter	SCFX002_d

Keine Betreuung durch andere Personen oder Einrichtungen	SCFX002_e


In [37]:
partner_mothers_care[partner_mothers_care["Woche"]==16]["childcare_vorher"].value_counts(normalize=True)

a    0.338736
b    0.225284
c    0.217180
e    0.116694
d    0.102107
Name: childcare_vorher, dtype: float64

In [38]:
partner_fathers_care[partner_fathers_care["Woche"]==16]["childcare_vorher"].value_counts(normalize=True)

a    0.331370
b    0.253314
c    0.220913
e    0.132548
d    0.061856
Name: childcare_vorher, dtype: float64

In [39]:
single_mothers_care[single_mothers_care["Woche"]==16]["childcare_vorher"].value_counts(normalize=True)

c    0.271429
a    0.257143
b    0.228571
d    0.214286
e    0.028571
Name: childcare_vorher, dtype: float64

In [40]:
#share of partners with kids in some form of daycare

In [41]:
partner_mothers_care[partner_mothers_care["Woche"]==16]["childcare_vorher"].value_counts(normalize=True)[["a","b","c"]].sum()

0.7811993517017828

In [42]:
partner_fathers_care[partner_fathers_care["Woche"]==16]["childcare_vorher"].value_counts(normalize=True)[["a","b","c"]].sum()

0.8055964653902797

In [43]:
single_mothers_care[single_mothers_care["Woche"]==16]["childcare_vorher"].value_counts(normalize=True)[["a","b","c"]].sum()

0.7571428571428571

*Nachher*

Durch mich selber	SCFX003_a

Durch die Notfallbetreuung der Kita / Schule	SCFX003_b

Durch eine andere Person im Haushalt (Alter 16-59)	SCFX003_c

Durch eine Person (Alter 16-59), die nicht im Haushalt lebt	SCFX003_d

Durch eine andere Person im Haushalt (Alter 60+)	SCFX003_e

Durch eine Person (Alter 60+), die nicht im Haushalt lebt	SCFX003_f

Das Kind kann/die Kinder können heute von keiner Person (Alter 16+) betreut werden. 	SCFX003_g

In der Kita oder Schule	SCFX003_h


In [44]:
partner_mothers_care[partner_mothers_care["Woche"]==16]["childcare_nachher"].value_counts(normalize=True)

a    0.482982
h    0.264182
c    0.175041
d    0.042139
f    0.014587
e    0.012966
g    0.008104
Name: childcare_nachher, dtype: float64

In [45]:
partner_mothers_care[partner_mothers_care["Woche"]==16]["childcare_nachher"].value_counts(normalize=True)[["h"]]

h    0.264182
Name: childcare_nachher, dtype: float64

In [46]:
partner_fathers_care[partner_fathers_care["Woche"]==16]["childcare_nachher"].value_counts(normalize=True)

a    0.379971
c    0.357879
h    0.195876
d    0.029455
f    0.017673
g    0.010309
e    0.008837
Name: childcare_nachher, dtype: float64

In [47]:
partner_fathers_care[partner_fathers_care["Woche"]==16]["childcare_nachher"].value_counts(normalize=True)[["h"]]

h    0.195876
Name: childcare_nachher, dtype: float64

In [48]:
single_mothers_care[single_mothers_care["Woche"]==16]["childcare_nachher"].value_counts(normalize=True)

a    0.442857
h    0.214286
c    0.142857
d    0.071429
e    0.057143
f    0.042857
g    0.028571
Name: childcare_nachher, dtype: float64

In [49]:
single_mothers_care[single_mothers_care["Woche"]==16]["childcare_nachher"].value_counts(normalize=True)[["h"]]

h    0.214286
Name: childcare_nachher, dtype: float64

## Results, Part4
* We see that actually more mothers in partnerships do the daycare than single mothers (48% vs. 44%)
* For single mothers, grandparents or people outside the household have been a crucial part (13%)
* Partnered fathers help at least in over a third of the cases (38%)